orchard_generators.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+import sys; assert sys.version_info[0] >= 3, "Python 3 required."
+
+from pyblake2 import blake2s
+
+from tv_output import render_args, render_tv
+from orchard_sinsemilla import group_hash, sinsemilla_hash_to_point
+
+# https://zips.z.cash/protocol/nu5.pdf#concretespendauthsig
+SPENDING_KEY_BASE = group_hash(b'z.cash:Orchard', b'G')
+
+# https://zips.z.cash/protocol/nu5.pdf#commitmentsandnullifiers
+NULLIFIER_K_BASE = group_hash(b'z.cash:Orchard', b'K')
+
+# https://zips.z.cash/protocol/nu5.pdf#concretehomomorphiccommit
+VALUE_COMMITMENT_VALUE_BASE = group_hash(b'z.cash:Orchard-cv', b'v')
+VALUE_COMMITMENT_RANDOMNESS_BASE = group_hash(b'z.cash:Orchard-cv', b'r')
+
+# Used in SinsemillaCommit (https://zips.z.cash/protocol/nu5.pdf#sinsemillacommitments)
+NOTE_COMMITMENT_BASE = group_hash(b'z.cash:Orchard-NoteCommit-r', b'')
+NOTE_COMMITMENT_Q = group_hash(b'z.cash:Orchard-NoteCommit-M', b'')
```

Comment:
The domain separator for SinsemillaHashToPoint is used as the message for GroupHash when deriving Q:
```suggestion
NOTE_COMMITMENT_Q = group_hash(b'z.cash:SinsemillaQ', b'z.cash:Orchard-NoteCommit-M')
```